rcap/cache_redis.go

```diff
@@ -0,0 +1,77 @@
+package rcap
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/pkg/errors"
+)
+
+type RedisCache struct {
+	config CacheConfig
+	client *redis.Client
+}
+
+type RedisConfig struct {
```

Comment:
I like how elegant it is that this configuration just flows in from the Directive